### PR TITLE
Index more dcp2 projects to `prod2` (#3752)

### PR DESCRIPTION
**Pull request by amarjandu**
https://github.com/DataBiosphere/azul/issues/3752

### Author

- [x] PR title references issue
- [x] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
- [x] Title of main commit references issue
- [x] PR is connected to Zenhub issue and description links to issue

### Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

### Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [ ] Sanity-checked history
- [ ] Pushed PR branch to Github
- [ ] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [ ] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Added PR reference to merge commit title
- [ ] Collected commit title tags in merge commit title
- [ ] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate       <sub>or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab
- [ ] Moved issues to `prod` or `Merged prod`               <sub>or this PR does not represent a promotion</sub>

### Operator (reindex) 

- [ ] Started reindex in target deployment                  <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures             <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment              <sub>or this PR does not require reindexing</sub>
- [ ] Filed backport PR                                     <sub>or this PR does not represent a hotfix</sub>

### Operator

- [ ] Delete `dcp10` indicies from `prod2` ES instance
- [ ] Unassigned PR

### CODE REVIEW

**Comment by amarjandu** [comment]
Ran `_select prod2 && make post_deploy` to confirm that the sources are accessible.

**Review by achave11 — COMMENTED** [review]
LGTM, approved!

**Review by achave11 — APPROVED** [review]

**Review by hannes-ucsc — COMMENTED** [review]
I blocked this issue (#3752) on #3767. There is no point in adding more snapshots without making sure the IT passes on all of them.

File: deployments/prod2/environment.py [hunk 1]
```diff
@@ -48,9 +48,203 @@ def mksrc(project, snapshot, subgraphs, ma: int = 0):
...
+    mksrc('datarepo-89f05580', 'hca_prod_dd7f24360c564709bd17e526bba4cc15__20220119_dcp2_20220121_dcp12', 37),
+    mksrc('datarepo-d6e13093', 'hca_prod_df88f39f01a84b5b92f43177d6c0f242__20220119_dcp2_20220121_dcp12', 1),
+    mksrc('datarepo-cd37664c', 'hca_prod_e0c74c7a20a445059cf138dcdd23011b__20220119_dcp2_20220121_dcp12', 2),
+    mksrc('datarepo-921c7df9', 'hca_prod_e526d91dcf3a44cb80c5fd7676b55a1d__20220119_dcp2_20220121_dcp12', 606),
+    mksrc('datarepo-8d441277', 'hca_prod_e57dc176ab98446b90c289e0842152fd__20220119_dcp2_20220121_dcp12', 94),
+    mksrc('datarepo-a2bba34f', 'hca_prod_e5d455791f5b48c3b568320d93e7ca72__20220119_dcp2_20220121_dcp12', 8),
+    mksrc('datarepo-32d08de8', 'hca_prod_e77fed30959d4fadbc15a0a5a85c21d2__20220119_dcp2_20220121_dcp12', 333),
+    mksrc('datarepo-f144a7cd', 'hca_prod_f83165c5e2ea4d15a5cf33f3550bffde__20220111_dcp1_20220126_dcp1', 15257),
+    mksrc('datarepo-224af5ae', 'hca_prod_e0009214c0a04a7b96e2d6a83e966ce0__20220113_dcp1_20220126_dcp1', 99840),
+    mksrc('datarepo-319b223a', 'hca_prod_e0009214c0a04a7b96e2d6a83e966ce0__20220119_dcp2_20220126_dcp12', 99841),
```
Comment (amarjandu) [inline deployments/prod2/environment.py:247]: Confirm with Hannes the last 3 snapshots, 2 from dcp1, and the last one that looks like it succeeds the second to last one.

Comment (amarjandu) [reply deployments/prod2/environment.py:247]: Also any `ma` snapshots in the list? Should there be 2 like in dev?

File: deployments/prod2/environment.py [hunk 2]
```diff
@@ -48,9 +48,203 @@ def mksrc(project, snapshot, subgraphs, ma: int = 0):
...
+    mksrc('datarepo-62e87fe3', 'hca_prod_c5f4661568de4cf4bbc2a0ae10f08243__20220118_dcp2_20220121_dcp12', 1),
+    mksrc('datarepo-44df5b5a', 'hca_prod_f48e7c39cc6740559d79bc437892840c__20220119_dcp2_20220121_dcp12', 14),
+    mksrc('datarepo-eb93ad96', 'hca_prod_f6133d2a9f3d4ef99c19c23d6c7e6cc0__20220119_dcp2_20220121_dcp12', 384),
+    mksrc('datarepo-e3c29b0f', 'hca_prod_f81efc039f564354aabb6ce819c3d414__20220119_dcp2_20220121_dcp12', 4),
+    mksrc('datarepo-11942c76', 'hca_prod_f83165c5e2ea4d15a5cf33f3550bffde__20220119_dcp2_20220121_dcp12', 7665),
+    mksrc('datarepo-c64a357d', 'hca_prod_f86f1ab41fbb4510ae353ffd752d4dfc__20220119_dcp2_20220121_dcp12', 20),
+    mksrc('datarepo-4167b729', 'hca_prod_f8aa201c4ff145a4890e840d63459ca2__20220119_dcp2_20220121_dcp12', 384),
+    mksrc('datarepo-590e9f21', 'hca_prod_faeedcb0e0464be7b1ad80a3eeabb066__20220119_dcp2_20220121_dcp12', 62),
+    mksrc('datarepo-f680e590', 'hca_prod_962bd805eb894c54bad2008e497d1307__20220118_dcp2_20220121_dcp12', 28),
+    mksrc('datarepo-9b7aa7dd', 'hca_prod_99101928d9b14aafb759e97958ac7403__20220118_dcp2_20220121_dcp12', 1190),
```
Comment (amarjandu) [inline deployments/prod2/environment.py:190]: The equivalent snapshot in `dev` is marked as `ma`

File: deployments/prod2/environment.py [hunk 3]
```diff
@@ -48,9 +48,203 @@ def mksrc(project, snapshot, subgraphs, ma: int = 0):
...
+    mksrc('datarepo-8a2c2dfd', 'hca_prod_996120f9e84f409fa01e732ab58ca8b9__20220118_dcp2_20220121_dcp12', 26),
+    mksrc('datarepo-9385cdd8', 'hca_prod_9d97f01f9313416e9b07560f048b2350__20220118_dcp2_20220121_dcp12', 43),
+    mksrc('datarepo-ddcd2940', 'hca_prod_a004b1501c364af69bbd070c06dbc17d__20220118_dcp2_20220121_dcp12', 34),
```
Comment (amarjandu) [inline deployments/prod2/environment.py:193]: This one too

File: deployments/prod2/environment.py (hunk 1 above)
Comment (hannes-ucsc) [reply deployments/prod2/environment.py:247]: No and no.

The ones marked `ma` in dev aren't actually managed-access, we just pretend they do for testing purposes.

**Comment by amarjandu** [comment]
self note: shortened PR chain

**Review by hannes-ucsc — APPROVED** [review]

File: deployments/prod2/environment.py [hunk 4]
```diff
@@ -48,9 +48,203 @@ def mksrc(project, snapshot, subgraphs, ma: int = 0):
...
+    mksrc('datarepo-ee4df1a4', 'hca_prod_dc1a41f69e0942a6959e3be23db6da56__20220119_dcp2_20220121_dcp12', 5),
+    mksrc('datarepo-89f05580', 'hca_prod_dd7f24360c564709bd17e526bba4cc15__20220119_dcp2_20220121_dcp12', 37),
+    mksrc('datarepo-d6e13093', 'hca_prod_df88f39f01a84b5b92f43177d6c0f242__20220119_dcp2_20220121_dcp12', 1),
+    mksrc('datarepo-cd37664c', 'hca_prod_e0c74c7a20a445059cf138dcdd23011b__20220119_dcp2_20220121_dcp12', 2),
+    mksrc('datarepo-921c7df9', 'hca_prod_e526d91dcf3a44cb80c5fd7676b55a1d__20220119_dcp2_20220121_dcp12', 606),
+    mksrc('datarepo-8d441277', 'hca_prod_e57dc176ab98446b90c289e0842152fd__20220119_dcp2_20220121_dcp12', 94),
+    mksrc('datarepo-a2bba34f', 'hca_prod_e5d455791f5b48c3b568320d93e7ca72__20220119_dcp2_20220121_dcp12', 8),
+    mksrc('datarepo-32d08de8', 'hca_prod_e77fed30959d4fadbc15a0a5a85c21d2__20220119_dcp2_20220121_dcp12', 333),
+    mksrc('datarepo-f144a7cd', 'hca_prod_f83165c5e2ea4d15a5cf33f3550bffde__20220111_dcp1_20220126_dcp1', 15257),
+    mksrc('datarepo-224af5ae', 'hca_prod_e0009214c0a04a7b96e2d6a83e966ce0__20220113_dcp1_20220126_dcp1', 99840),
```
Comment (hannes-ucsc) [inline deployments/prod2/environment.py:246]: Wrong catalog. Note the qualifiers in the name.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
The way you structured the checklist, you'd be deleting the dcp10 catalog after you reindex. That puts unnecessary strain on the ES instance. You should delete the old catalog shortly before you reindex. That should be covered in the manual.

Also follow this [thread](https://humancellatlas.slack.com/archives/C01360XN04S/p1643852281038889?thread_ts=1641323990.260200&cid=C01360XN04S). We may be getting another snapshot.

File: deployments/prod2/environment.py (hunk 4 above)
Comment (amarjandu) [reply deployments/prod2/environment.py]: Removed the two `dcp1` sources, added the source that was mentioned in the Slack thread. 195 total sources added, confirmed access with `make post_deploy`.

Comment (hannes-ucsc) [reply deployments/prod2/environment.py]: These two need to be put in the right catalog.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
https://github.com/DataBiosphere/azul/pull/3783#discussion_r798982057

and ticket needs to be updated to reflect the added row.

File: deployments/prod2/environment.py (hunk 4 above)
Comment (amarjandu) [reply deployments/prod2/environment.py]: Ah I misunderstood, i thought it was only `dcp2` projects, I'll add them to the `dcp1` catalog in a new commit in this PR.

**Comment by amarjandu** [comment]
Added rows 217, 218 to `dcp1` and updated issue description

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
fixup! missing from one of the commit titles.

Also, we're still missing some dcp1 snapshots on the spreadsheet.

https://humancellatlas.slack.com/archives/C01360XN04S/p1643935150634639?thread_ts=1641323990.260200&cid=C01360XN04S

**Comment by amarjandu** [comment]
>fixup! missing from one of the commit titles.

It seemed incorrect to add the changes to the `dcp1` sources to the `dcp2` commit, so there are two commits that are: 

[r] Index more dcp2 projects to `prod2` (#3752) <- for `dcp2` source changes
[r] Index more dcp1 projects to `prod2` (#3752) <- for `dcp1` source changes

It seemed appropriate to track the changes in this way.

> Also, we're still missing some dcp1 snapshots on the spreadsheet.

I'll add fixups for these.

**Comment by achave11** [comment]
Added missing snapshots to DCP1 as per the update [in Slack](https://humancellatlas.slack.com/archives/C01360XN04S/p1643987229638549?thread_ts=1641323990.260200&cid=C01360XN04S).

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
@amarjandu I didn't notice the dcp1/dcp2 difference. 

@achave11 Since you two collaborated, this would have to be structured as a split commit.

**Comment by achave11** [comment]
IT succeeded. 
Screen Shot 2022-02-15 at 9 26 32 AM

**Pull request merged**